lib/services/local/integrations.go

```diff
@@ -36,11 +36,43 @@ const (
 
 // IntegrationsService manages Integrations in the Backend.
 type IntegrationsService struct {
-	svc generic.Service[types.Integration]
+	svc              generic.Service[types.Integration]
+	backend          backend.Backend
+	preDeleteChecks  []PreDeleteCheck
```

Comment:
Is this intended to be called for any deletion in the `IntegrationsService`, or just the deletion of OIDC integrations? What happens if we extend the scope of `IntegrationsService` to multiple types of integrations, for example? Why should the caller be responsible for figuring out the interdependency between resources managed by the two services?

 Is this really better than spawning a new `ExternalAuditStorageService` on the fly, depending on the same "cache mode" that's enabled if `deleteAllEnabled` is set?

Reply:
I reworked this with the AtomicWrite rewrite